src/Functional.Primitives.FluentAssertions/Functional.Primitives.FluentAssertions.Tests/ResultTypeAssertionsTests.cs

```diff
@@ -8,74 +8,66 @@ public class ResultTypeAssertionsTests
 	{
 		public class SuccessChecks
 		{
-			public class WithNoSpecialAssertions
-			{
-				[Fact]
-				public void ShouldNotThrowException() => new Action(() => Result.Success<int, Exception>(3).Should().BeSuccessful()).Should().NotThrow();
-
-				[Fact]
-				public void ShouldThrowException() => new Action(() => Result.Failure<int, Exception>(new Exception()).Should().BeSuccessful()).Should().Throw<Exception>();
-			}
+			private const int VALUE = 3;
 
-			public class WithExpectedValueAssertion
+			[Fact]
+			public void ShouldNotThrowException() => new Action(() =>
 			{
-				[Fact]
-				public void ShouldNotThrowException() => new Action(() => Result.Success<int, Exception>(3).Should().BeSuccessfulWithExpectedValue(3)).Should().NotThrow();
-
-				[Fact]
-				public void ShouldThrowExceptionIfNotExpectedSuccessValue() => new Action(() => Result.Success<int, Exception>(2).Should().BeSuccessfulWithExpectedValue(30)).Should().Throw<Exception>();
+				Result.Success<int, Exception>(VALUE)
+					.Should()
+					.BeSuccessful("an exception occurred");
 
-				[Fact]
-				public void ShouldThrowExceptionIfFaultedResult() => new Action(() => Result.Failure<int, Exception>(new Exception()).Should().BeSuccessfulWithExpectedValue(3)).Should().Throw<Exception>();
-			}
+			}).Should().NotThrow();
 
-			public class WithAssertionsToMatchProperties
+			[Fact]
+			public void ShouldNotThrowExceptionWhenAdditionalAssertionSucceeds() => new Action(() =>
 			{
-				[Fact]
-				public void ShouldNotThrowException() => new Action(() => Result.Success<int, Exception>(3).Should().BeSuccessful(value => value.Should().BeGreaterOrEqualTo(0))).Should().NotThrow();
+				Result.Success<int, Exception>(VALUE)
+					.Should()
+					.BeSuccessful()
+					.AndValue
```

Comment:
I am currently on the fence between sticking with `AndValue` or making the individual result cases more explicit in the assertion syntax...

```
// something like...
.Should().BeSuccessful().AndSuccessValue.Should().Be(VALUE);
.Should().BeFaulted().AndFailureValue.Should().Be(...);
```

Thoughts @ChrisDusyk , @willikev , @JohannesMoersch ?

Reply:
I'm leaning towards the argument that `.AndSuccessValue` and `.AndFailureValue` would be more true to the current version of the `Functional` API.

Looking at the README of https://github.com/JohannesMoersch/Functional:'

- `Option<T>` has `HasValue()` and `ValueOrDefault()` methods
- `Result<TSuccess, TFailure>` does _not_ have any associated methods or public properties with `Value` in their name
- `Result<TSuccess, TFailure>` _does_ have a `Success()` method that returns an `Option<TSuccess>`, _and_ a `Failure()` method that returns an `Option<TFailure>`.
  - As mentioned before, those `Option`s _do_ have `ValueOrDefault()` methods.

So I would assume that `.AndSuccessValue.Should()....` would let me assert directly on the `TSuccess`, and that `.AndFailureValue.Should()....` would let me assert directly on the `TFailure`, bypassing any `Option<T>` syntax.

---

Granted, I'd _also_ assume that `.Should().BeSuccessful().AndValue.Should()....` would let me assert directly on the `TSuccess`, and that `.Should().BeFaulted().AndValue.Should()....` would let me assert directly on the `TFailure`. So if we wanted to keep this syntax concise, I'd be okay with sticking with `AndValue`.

Reply:
In updating `Functional.CQS` and resolving warnings associated with `Obsolete` methods, I came across a few instances of this:
```
fromJson.Should().BeSuccessful().AndValue.Should().HaveValue().AndValue.IsLike(obj));
```

`AndValue` is fine when there is no nesting of `Option<T>`, `Result<TSuccess, TFailure>` types.  Once you start dealing with nested types the assertions become difficult to read / reason about, which defeats the point of the `FluentAssertions` part of the package name.  I will go ahead with a change to move off `AndValue`.